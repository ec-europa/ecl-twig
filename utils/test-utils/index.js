const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
