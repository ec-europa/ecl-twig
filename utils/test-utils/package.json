{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
