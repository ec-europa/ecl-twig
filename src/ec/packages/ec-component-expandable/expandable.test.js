import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

// Add SVG icon path.
demoData.button.icon.path = 'example';

describe('EC - Expandable', () => {
  const template = path.resolve(__dirname, './expandable.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test('renders correctly when expanded', () => {
    expect.assertions(1);

    const expanded = Object.assign({}, demoData, { expanded: true });
    return expect(render(expanded)).resolves.toMatchSnapshot();
  });
});
