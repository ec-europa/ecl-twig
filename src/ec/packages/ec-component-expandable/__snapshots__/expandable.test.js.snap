// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Expandable renders correctly 1`] = `
<div
  class="ecl-expandable"
  data-ecl-expandable="true"
>
  <button
    aria-controls="expandable-example-content"
    class="ecl-button ecl-button--secondary ecl-expandable__toggle"
    data-ecl-expandable-toggle="true"
    data-ecl-label-collapsed="Collapsed button"
    data-ecl-label-expanded="Expanded button"
    type="button"
  >
    <span
      class="ecl-button__container"
    >
      <span
        class="ecl-button__label"
        data-ecl-label=""
      >
        Collapsed button
      </span>
      <svg
        aria-controls="expandable-example-content"
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
        data-ecl-expandable-toggle="true"
        data-ecl-label-collapsed="Collapsed button"
        data-ecl-label-expanded="Expanded button"
        focusable="false"
      >
        <use
          xlink:href="example#ui--corner-arrow"
        />
      </svg>
    </span>
  </button>
  <div
    class="ecl-expandable__content"
    hidden=""
    id="expandable-example-content"
  >
    <p
      class="ecl-u-type-paragraph-m"
    >
      The EU is building an energy union that ensures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
    </p>
  </div>
</div>
`;

exports[`EC - Expandable renders correctly when expanded 1`] = `
<div
  class="ecl-expandable"
  data-ecl-expandable="true"
>
  <button
    aria-controls="expandable-example-content"
    aria-expanded="true"
    class="ecl-button ecl-button--secondary ecl-expandable__toggle"
    data-ecl-expandable-toggle="true"
    data-ecl-label-collapsed="Collapsed button"
    data-ecl-label-expanded="Expanded button"
    type="button"
  >
    <span
      class="ecl-button__container"
    >
      <span
        class="ecl-button__label"
        data-ecl-label=""
      >
        Collapsed button
      </span>
      <svg
        aria-controls="expandable-example-content"
        aria-expanded="true"
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-icon--rotate-180 ecl-button__icon ecl-button__icon--after"
        data-ecl-expandable-toggle="true"
        data-ecl-label-collapsed="Collapsed button"
        data-ecl-label-expanded="Expanded button"
        focusable="false"
      >
        <use
          xlink:href="example#ui--corner-arrow"
        />
      </svg>
    </span>
  </button>
  <div
    class="ecl-expandable__content"
    id="expandable-example-content"
  >
    <p
      class="ecl-u-type-paragraph-m"
    >
      The EU is building an energy union that ensures Europe’s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
    </p>
  </div>
</div>
`;
